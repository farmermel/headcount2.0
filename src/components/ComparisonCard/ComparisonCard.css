.comparison-card {
  justify-content: space-evenly;
  text-align: center;
}

.ratio {
  font-size: 2em;
}