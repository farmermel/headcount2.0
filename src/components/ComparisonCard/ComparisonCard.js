import React from 'react';
import PropTypes from 'prop-types';
import './ComparisonCard.css';

const ComparisonCard = ({ districts, comparativeAnalysis}) => {    
  const district1key = Object.keys(districts)[0];
  const district2key = Object.keys(districts)[1];
  const district1 = district1key ? `: ${districts[district1key].avg}` : 'Pick first card to compare';
  const district2 = district2key ? `: ${districts[district2key].avg}` : 'Pick second card to compare';
  const analysis = comparativeAnalysis.compared ? comparativeAnalysis.compared : ''
  
  return (
    <div className="card comparison-card">
      <h3>{district1key} {district1}</h3>
      <h3>{analysis}</h3>
      <h3>{district2key} {district2}</h3>
    </div>
  );
};

ComparisonCard.propTypes = {
  districts: PropTypes.objectOf(PropTypes.shape({
    avg: PropTypes.number,
    data: PropTypes.objectOf(PropTypes.number),
    location: PropTypes.string
  })),
  comparativeAnalysis: PropTypes.objectOf(PropTypes.number)
}

export default ComparisonCard;