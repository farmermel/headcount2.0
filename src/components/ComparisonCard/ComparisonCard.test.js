import React from 'react';
import ReactDOM from 'react-dom';
import ComparisonCard from './ComparisonCard.js';
import { shallow } from 'enzyme';

describe('ComparisonCard', () => {
  let wrapper;
  let mockDistricts = {
    COLORADO: {
      avg: 0.4,
      data: {
        2004: 0.4,
        2005: 0.8
      },
      location: 'COLORADO'
    },
    ASPEN: {
      avg: 0.6,
      data: {
        2004: 0.6,
        2005: 0.9
      },
      location: 'ASPEN 20'
    }
  };
  let mockComparativeAnalysis = {
    'ACADEMY 20': 0.407,
    'YUMA SCHOOL DISTRICT 1': 0.909,
    compared: 0.448
  };

  it('should match snapshot', () => {
    wrapper = shallow(
      <ComparisonCard
        comparativeAnalysis={mockComparativeAnalysis}
        districts={mockDistricts}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
