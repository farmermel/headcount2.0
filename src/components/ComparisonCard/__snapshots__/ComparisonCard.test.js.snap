// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonCard should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonCard
    comparativeAnalysis={
        Object {
            "ACADEMY 20": 0.407,
            "YUMA SCHOOL DISTRICT 1": 0.909,
            "compared": 0.448,
          }
    }
    districts={
        Object {
            "ASPEN": Object {
              "avg": 0.6,
              "data": Object {
                "2004": 0.6,
                "2005": 0.9,
              },
              "location": "ASPEN 20",
            },
            "COLORADO": Object {
              "avg": 0.4,
              "data": Object {
                "2004": 0.4,
                "2005": 0.8,
              },
              "location": "COLORADO",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          COLORADO
           
          : 0.4
</h3>,
        <h3>
          0.448
</h3>,
        <h3>
          ASPEN
           
          : 0.6
</h3>,
      ],
      "className": "card comparison-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "COLORADO",
            " ",
            ": 0.4",
          ],
        },
        "ref": null,
        "rendered": Array [
          "COLORADO",
          " ",
          ": 0.4",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0.448,
        },
        "ref": null,
        "rendered": 0.448,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "ASPEN",
            " ",
            ": 0.6",
          ],
        },
        "ref": null,
        "rendered": Array [
          "ASPEN",
          " ",
          ": 0.6",
        ],
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            COLORADO
             
            : 0.4
</h3>,
          <h3>
            0.448
</h3>,
          <h3>
            ASPEN
             
            : 0.6
</h3>,
        ],
        "className": "card comparison-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "COLORADO",
              " ",
              ": 0.4",
            ],
          },
          "ref": null,
          "rendered": Array [
            "COLORADO",
            " ",
            ": 0.4",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0.448,
          },
          "ref": null,
          "rendered": 0.448,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "ASPEN",
              " ",
              ": 0.6",
            ],
          },
          "ref": null,
          "rendered": Array [
            "ASPEN",
            " ",
            ": 0.6",
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
